### PR TITLE
[move] Add Signer module and native function

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- Added 0x0::Signer module
- Added native fun borrow_address
- Added user-land fun address_of

## Test Plan

- new tests
- updated old tests

### CODE REVIEW

File: language/ir-testsuite/tests/signer/borrow_address.mvir [hunk 1]
```diff
@@ -0,0 +1,11 @@
+//! account: alice, 90000
```
Comment (sblackshear) [inline language/ir-testsuite/tests/signer/borrow_address.mvir:1]: Nit: `, 9000` not needed

File: language/ir-testsuite/tests/signer/borrow_address.mvir [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+//! new-transaction
+//! sender: alice
+
+import 0x0.Signer;
+main(s: &signer) {
+    let x: address;
+    x = {{alice}};
+    assert(Signer.borrow_address(move(s)) == &x, 42);
+    return;
+}
```
Comment (sblackshear) [inline language/ir-testsuite/tests/signer/borrow_address.mvir:11]: Do we need both Move and mvir tests for these? I have found it painful to maintain two copies of the same test when we do this elsewhere...

File: language/stdlib/modules/signer.move [hunk 3]
```diff
@@ -0,0 +1,9 @@
+address 0x0 {
+module Signer {
+    native public fun borrow_address(s: &signer): &address;
```
Comment (sblackshear) [inline language/stdlib/modules/signer.move:3]: Nit: comments

File: language/ir-testsuite/tests/signer/borrow_address.mvir (hunk 2 above)
Comment (tnowacki) [reply language/ir-testsuite/tests/signer/borrow_address.mvir:11]: I could move them just to Move, sure

**Pull request closed without merging**